File: unknownbank_flutter/lib/models/credit_card_application_data.dart
class CreditCardApplicationData {
  final String title;
  final String fullName;
  final String icNumber;
  final String dateOfBirth;
  final String email;
  final String phoneCountryCode;
  final String phoneNumber;

  final String nameOnCard;
  final String education;
  final String gender;
  final String race;
  final String maritalStatus;
  final String motherName;

  final String employerName;
  final String occupation;
  final String sector;
  final String employmentType;
  final String businessClassification;
  final String yearsOfService;
  final String monthsOfService;

  final String monthlyNetIncome;
  final String monthlyCommitments;
  final String retirementIncomeSource;
  final String statementDelivery;
  final String collectionState;
  final String collectionDistrict;
  final String collectionBranch;

  final String nricFrontDocument;
  final String nricBackDocument;
  final String salaryDocument;
  final List<String> additionalDocuments;

  const CreditCardApplicationData({
    this.title = 'MS / PUAN',
    this.fullName = 'Farah Amira Ali',
    this.icNumber = '851010145640',
    this.dateOfBirth = '10-10-1985',
    this.email = 'farah.amira@gmail.com',
    this.phoneCountryCode = '+60',
    this.phoneNumber = '122500440',

    this.nameOnCard = 'Farah Amira Ali',
    this.education = 'Degree',
    this.gender = 'Female',
    this.race = 'Malay',
    this.maritalStatus = 'Married',
    this.motherName = 'Dayang',

    this.employerName = 'Star Trading',
    this.occupation = 'Sales Manager',
    this.sector = 'Logistic',
    this.employmentType = 'Private Employed',
    this.businessClassification = 'Private Limited',
    this.yearsOfService = '10',
    this.monthsOfService = '6',

    this.monthlyNetIncome = '12,000',
    this.monthlyCommitments = '5,860',
    this.retirementIncomeSource = 'Rental Income',
    this.statementDelivery = 'Email',
    this.collectionState =
        'Wilayah Persekutuan Kuala Lumpur',
    this.collectionDistrict = 'Kuala Lumpur',
    this.collectionBranch = 'Bangsar',

    this.nricFrontDocument =
        'farah_amira_ID_front.pdf',
    this.nricBackDocument =
        'farah_amira_ID_back.pdf',
    this.salaryDocument =
        'farah_amira_PaySlip.pdf',
    this.additionalDocuments = const [],
  });

  CreditCardApplicationData copyWith({
    String? title,
    String? fullName,
    String? icNumber,
    String? dateOfBirth,
    String? email,
    String? phoneCountryCode,
    String? phoneNumber,

    String? nameOnCard,
    String? education,
    String? gender,
    String? race,
    String? maritalStatus,
    String? motherName,

    String? employerName,
    String? occupation,
    String? sector,
    String? employmentType,
    String? businessClassification,
    String? yearsOfService,
    String? monthsOfService,

    String? monthlyNetIncome,
    String? monthlyCommitments,
    String? retirementIncomeSource,
    String? statementDelivery,
    String? collectionState,
    String? collectionDistrict,
    String? collectionBranch,

    String? nricFrontDocument,
    String? nricBackDocument,
    String? salaryDocument,
    List<String>? additionalDocuments,
  }) {
    return CreditCardApplicationData(
      title: title ?? this.title,
      fullName: fullName ?? this.fullName,
      icNumber: icNumber ?? this.icNumber,
      dateOfBirth:
          dateOfBirth ?? this.dateOfBirth,
      email: email ?? this.email,
      phoneCountryCode:
          phoneCountryCode ?? this.phoneCountryCode,
      phoneNumber:
          phoneNumber ?? this.phoneNumber,

      nameOnCard:
          nameOnCard ?? this.nameOnCard,
      education:
          education ?? this.education,
      gender: gender ?? this.gender,
      race: race ?? this.race,
      maritalStatus:
          maritalStatus ?? this.maritalStatus,
      motherName:
          motherName ?? this.motherName,

      employerName:
          employerName ?? this.employerName,
      occupation:
          occupation ?? this.occupation,
      sector: sector ?? this.sector,
      employmentType:
          employmentType ?? this.employmentType,
      businessClassification:
          businessClassification ??
              this.businessClassification,
      yearsOfService:
          yearsOfService ?? this.yearsOfService,
      monthsOfService:
          monthsOfService ?? this.monthsOfService,

      monthlyNetIncome:
          monthlyNetIncome ?? this.monthlyNetIncome,
      monthlyCommitments:
          monthlyCommitments ??
              this.monthlyCommitments,
      retirementIncomeSource:
          retirementIncomeSource ??
              this.retirementIncomeSource,
      statementDelivery:
          statementDelivery ??
              this.statementDelivery,
      collectionState:
          collectionState ?? this.collectionState,
      collectionDistrict:
          collectionDistrict ??
              this.collectionDistrict,
      collectionBranch:
          collectionBranch ??
              this.collectionBranch,

      nricFrontDocument:
          nricFrontDocument ??
              this.nricFrontDocument,
      nricBackDocument:
          nricBackDocument ??
              this.nricBackDocument,
      salaryDocument:
          salaryDocument ?? this.salaryDocument,
      additionalDocuments:
          additionalDocuments ??
              this.additionalDocuments,
    );
  }
}

File: Unknownbank/app/src/main/java/com/kasatria/kasatriaunknownbank/CreditCardApplicationData.kt
package com.kasatria.kasatriaunknownbank

data class CreditCardApplicationData(

    // Screen 07 — Personal Details
    val title: String = "MS / PUAN",
    val fullName: String = "Farah Amira Ali",
    val icNumber: String = "851010145640",
    val dateOfBirth: String = "10-10-1985",
    val email: String = "farah.amira@gmail.com",
    val phoneCountryCode: String = "+60",
    val phoneNumber: String = "122500440",

    // Screen 08 — About You
    val nameOnCard: String = "Farah Amira Ali",
    val education: String = "Degree",
    val gender: String = "Female",
    val race: String = "Malay",
    val maritalStatus: String = "Married",
    val motherName: String = "Dayang",

    // Screen 09 — Job Details
    val employerName: String = "Star Trading",
    val occupation: String = "Sales Manager",
    val sector: String = "Logistic",
    val employmentType: String = "Private Employed",
    val businessClassification: String = "Private Limited",
    val yearsOfService: String = "10",
    val monthsOfService: String = "6",

    // Screen 10 — Extra Details
    val monthlyNetIncome: String = "12,000",
    val monthlyCommitments: String = "5,860",
    val retirementIncomeSource: String = "Rental Income",
    val statementDelivery: String = "Email",
    val collectionState: String =
        "Wilayah Persekutuan Kuala Lumpur",
    val collectionDistrict: String = "Kuala Lumpur",
    val collectionBranch: String = "Bangsar",

    //Screen 13 - Credit Card Details
    val nricFrontDocument: String = "farah_amira_ID_front.pdf",
    val nricBackDocument: String = "farah_amira_ID_back.pdf",
    val salaryDocument: String = "farah_amira_PaySlip.pdf",
    val additionalDocuments: List<String> = emptyList()
)